### PR TITLE
D3ASIM-3529: Added group_id to KafkaConsumer in order to be able to l…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
…oad balance incomming results messages accross multiple simulation_results consumers.

### CODE REVIEW

File: d3a_interface/kafka_communication/kafka_consumer.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
     KAFKA_URL, DEFAULT_KAFKA_URL, KAFKA_USERNAME,
     KAFKA_PASSWORD, KAFKA_COMMUNICATION_SECURITY_PROTOCOL,
     KAFKA_SASL_AUTH_MECHANISM,
-    KAFKA_API_VERSION, create_kafka_new_ssl_context, KAFKA_RESULTS_TOPIC)
+    KAFKA_API_VERSION, create_kafka_new_ssl_context, KAFKA_RESULTS_TOPIC, KAFKA_RESULTS_GROUP_ID)
```
Comment (mclrch) [inline d3a_interface/kafka_communication/kafka_consumer.py:10]: What about change it to something more generic e.g: KAFKA_CONSUMER_GROUP_ID

**Review by spyrostz — APPROVED** [review]
Once @mclrch 's comment is dealt with LGTM. Also I think you need to increase the number of partitions in dev env's Kafka topic, for the results service to scale correctly.

**Comment by hannesdiedrich** [comment]
> Once @mclrch 's comment is dealt with LGTM. Also I think you need to increase the number of partitions in dev env's Kafka topic, for the results service to scale correctly.

I am not sure what you mean and I had a look into the Kafka documentation but did not find anything that suiting to your comment (a setting that sets the number of partitions) Can you please elaborate more ? Thanks!

**Comment by spyrostz** [comment]
> > Once @mclrch 's comment is dealt with LGTM. Also I think you need to increase the number of partitions in dev env's Kafka topic, for the results service to scale correctly.
> 
> I am not sure what you mean and I had a look into the Kafka documentation but did not find anything that suiting to your comment (a setting that sets the number of partitions) Can you please elaborate more ? Thanks!

Have a look at this for instance https://www.instaclustr.com/the-power-of-kafka-partitions-how-to-get-the-most-out-of-your-kafka-cluster/ . Setting the number of partitions is done when creating the Kafka topic, it is one of the parameters when you do that via CMak

**Pull request merged**